server/protocols/webadmin/webadmin-swagger/src/main/java/org/apache/james/webadmin/swagger/route/MailboxRoute.java

```diff
@@ -0,0 +1,127 @@
+ ****************************************************************/
+
+package org.apache.james.webadmin.swagger.route.test;
+
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+
+import org.apache.james.webadmin.swagger.model.ApiError;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiImplicitParams;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+
+@Api
+@Path("/users/{username}/mailboxes")
+@Produces("application/json")
+public class MailboxRoute implements Route {
+    @Override
+    public Object handle(Request request, Response response) throws Exception {
+        return null;
+    }
+
+    @POST
+    @Path("/{mailboxName}")
+    @ApiOperation(value = "Create a mailbox of the selected user.", nickname="CreateUserMailbox")
+    @ApiImplicitParams({
+            @ApiImplicitParam(required = true, dataType = "string", name = "username", paramType = "path"),
+            @ApiImplicitParam(required = true, dataType = "string", name = "mailboxName", paramType = "path")
+    })
+    @ApiResponses(value = {
+            @ApiResponse(code = 204, message = "OK. The mailbox now exists on the server.", response=String.class),
+            @ApiResponse(code = 400, message = "Invalid mailbox name", response=ApiError.class),
+            @ApiResponse(code = 401, message = "Unauthorized. The user is not authenticated on the platform", response=ApiError.class),
+            @ApiResponse(code = 404, message = "The user name does not exist.", response=ApiError.class),
+            @ApiResponse(code = 500, message = "Internal server error - Something went bad on the server side.", response=ApiError.class)
+    })
+    public void createUserMailbox() {
+    }
+
+    @GET
+    @Path("/{mailboxName}")
+    @ApiImplicitParams({
+            @ApiImplicitParam(required = true, dataType = "string", name = "username", paramType = "path"),
+            @ApiImplicitParam(required = true, dataType = "string", name = "mailboxName", paramType = "path")
+    })
+    @ApiOperation(value = "Testing existence of a mailbox.")
+    @ApiResponses(value = {
+            @ApiResponse(code = 204, message = "The mailbox exists", response=String.class),
+            @ApiResponse(code = 400, message = "Invalid mailbox name", response=ApiError.class),
+            @ApiResponse(code = 401, message = "Unauthorized. The user is not authenticated on the platform", response=ApiError.class),
+            @ApiResponse(code = 404, message = "The user name does not exist.", response=ApiError.class),
+            @ApiResponse(code = 500, message = "Internal server error - Something went bad on the server side.", response=ApiError.class)
+    })
+    public void mailboxIsExisted() {
```

Comment:
You should rather annotate the *Routes* object directly in webadmin modules.

This way, annotation would with route definition.

(You can for this extract the individual route definition in their own method in *define*, and then annotate these methods)